lib/core/sdam/topology.js

```diff
@@ -345,8 +345,14 @@ class Topology extends EventEmitter {
       if (typeof selector !== 'function') {
         options = selector;
 
-        translateReadPreference(options);
-        const readPreference = options.readPreference || ReadPreference.primary;
+        let readPreference;
+        if (selector instanceof ReadPreference) {
```

Comment:
Is the idea here that you could invoke selectServer like ```topology.selectServer(readPreference, callback)```?

Unless this is dictated by the spec, I'd prefer to avoid this type of API b/c there are other relevant fields to server selection. Can we instead include the `readPreference` in the `options` object?